datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemRadioGroupViewHolderFactory.kt

```diff
@@ -60,18 +60,14 @@ internal object QuestionnaireItemRadioGroupViewHolderFactory :
         radioGroup.setOnCheckedChangeListener(null)
         var index = 0
         questionnaireItemViewItem.answerOption.forEach {
-          radioGroup.addView(
-            RadioButton(radioGroup.context, null, R.attr.radioButtonStyleQuestionnaire).apply {
-              id = index++ // Use the answer option index as radio button ID
-              text = it.displayString
-              layoutParams =
-                ViewGroup.LayoutParams(
-                  ViewGroup.LayoutParams.MATCH_PARENT,
-                  ViewGroup.LayoutParams.WRAP_CONTENT
-                )
-              isChecked = it.valueCoding.equalsDeep(answer)
-            }
-          )
+          val radioButtonItem =
+            LayoutInflater.from(radioGroup.context)
+              .inflate(R.layout.questionnaire_item_radio_button, null)
+          val radioButton = radioButtonItem.findViewById<RadioButton>(R.id.radio_button)
+          radioButton.isChecked = it.valueCoding.equalsDeep(answer)
+          radioButton.text = it.displayString
+          radioButton.id = index++
```

Comment:
i don't know if this id is used to clear the radio button selection. but it would be better to use android api to generate the id rather than using the index value.

```suggestion
          radioButton.id = View.generateViewId()
```